src/main/java/org/jclouds/vcloud/director/v1_5/compute/strategy/VCloudDirectorComputeServiceAdapter.java

```diff
@@ -676,16 +675,21 @@ private Vdc getVdc(String locationId) {
    }
 
    private Set<Vm> getAvailableVMsFromVAppTemplate(VAppTemplate vAppTemplate) {
-      return ImmutableSet.copyOf(filter(vAppTemplate.getChildren(), new Predicate<Vm>() {
-         // filter out vms in the vApp template with computer name that contains underscores, dots,
-         // or both.
-         @Override
-         public boolean apply(Vm input) {
-            GuestCustomizationSection guestCustomizationSection = api.getVmApi().getGuestCustomizationSection(input.getId());
-            String computerName = guestCustomizationSection.getComputerName();
-            return computerName.equals(computerName);
-         }
-      }));
+      // FIXME What should this be? Previously it did `computerName.equals(computerName)`,
```

Comment:
Here is the conversation we had on another channel:

@aledsage wrote:
@andreaturli (cc @bostko) jclouds 2.0.0 uses `error-prone` for some good (and some extremely strict) checks. It spotted this problem: https://github.com/cloudsoft/jclouds-vcloud-director/blob/release/1.9.3/src/main/java/org/jclouds/vcloud/director/v1_5/compute/strategy/VCloudDirectorComputeServiceAdapter.java#L686 (`computerName` is compared with itself). What should this code be? Should it really "filter out vms in the vApp template with computer name that contains underscores, dots, or both" as the name suggests? Or is that no longer needed (given we've been running without it for quite a while!)?

@andreaturli wrote:
I’d implement the "filter out vms in the vApp template with computer name that contains underscores, dots, or both” but not sure if it is still useful — probably safer to change the impl and leave the method there, wdyt?

@aledsage wrote:
@andreaturli I'm not sure. It's been working for a customer with the existing behaviour (where the filter effectively does nothing). So I'm tempted to just remove the filter. Why do we need to remove things with "underscores, dots, or both"?

@andreaturli wrote:
if I remember right, names with underscore and dot are not valid `computerName`